packages/core/src/loaders/errors/YamlSyntaxErrorDetailsExtractor.ts

```diff
@@ -0,0 +1,26 @@
+import { BaseSyntaxErrorDetailsExtractor } from './BaseSyntaxErrorDetailsExtractor';
+import { YAMLException } from 'js-yaml';
+
+export class YamlSyntaxErrorDetailsExtractor extends BaseSyntaxErrorDetailsExtractor<YAMLException> {
+  private readonly LOCATION_PATTERN = /\((\d+):(\d+)\)$/;
+
+  protected extractMessage(error: YAMLException): string {
+    return error.message
+      .split('\n')[0]
+      .replace(/^YAMLException: /, '')
+      .replace(this.LOCATION_PATTERN, '')
```

Comment:
Let's introduce a local variable

```suggestion
    const [someVarName]: string[] = error.message.split('\n');
    
    return someVarName
      .replace(/^YAMLException: /, '')
      .replace(this.LOCATION_PATTERN, '')
```